### PR TITLE
Allow type mismatch for GCC 10+

### PR DESCRIPTION
**Pull request by ChristopherMayes**
GCC 10 gives the following errors:

```make
IMPACT-T/src/DataStruct/Data.f90:1464:23:

 1461 |         call MPI_BCAST(this%NxIntvRfgt,1,MPI_INTEGER,0,MPI_COMM_WORLD,ierr)
      |                       2
......
 1464 |         call MPI_BCAST(this%XminRfgt,1,MPI_DOUBLE_PRECISION,0,MPI_COMM_WORLD,ierr)
      |                       1
Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(8)/INTEGER(4)).
```
This adds the `-fallow-argument-mismatch` flag to CMakeLists.txt to allow this mismatch.

Additionally, the Readme.txt and Instructions are unified.

### CODE REVIEW

**Comment by ChristopherMayes** [comment]
@ax3l any comments?

**Comment by ChristopherMayes** [comment]
This was previously noticed in https://github.com/impact-lbl/IMPACT-T/pull/13, but that was prior to the CMake method and did not check the compiler.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
-# Using Anaconda
+
+# IMPACT-T
```
Comment (ax3l) [inline README.md:2]: The change in the README looks unrelated to the PR, can you please open this separately if this is needed?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -13,6 +13,15 @@ project(ImpactT LANGUAGES Fortran)
...
 
+# Allow type mismatch for GCC 10
+# (Adapted from from: https://github.com/Unidata/netcdf-fortran/blob/main/CMakeLists.txt)
+# Check to see if compiler supports
+# -fallow-argument-mismatch-flag introduced in gcc 10.
+INCLUDE (CheckFortranCompilerFlag)
+check_fortran_compiler_flag("-fallow-argument-mismatch" COMPILER_HAS_ALLOW_ARGUMENT_MISMATCH)
+IF(COMPILER_HAS_ALLOW_ARGUMENT_MISMATCH)
+  SET(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch")
+ENDIF(COMPILER_HAS_ALLOW_ARGUMENT_MISMATCH)
```
Comment (ax3l) [inline src/CMakeLists.txt]: Lowercase to fit surrounding style:
```suggestion
include(CheckFortranCompilerFlag)
check_fortran_compiler_flag("-fallow-argument-mismatch" COMPILER_HAS_ALLOW_ARGUMENT_MISMATCH)
if(COMPILER_HAS_ALLOW_ARGUMENT_MISMATCH)
  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch")
endif(COMPILER_HAS_ALLOW_ARGUMENT_MISMATCH)
```

File: README.md (hunk 1 above)
Comment (ChristopherMayes) [reply README.md:2]: I could revert, but is there anything against it?

**Comment by ax3l** [comment]
Instead of adding to ignore the warning, is it possible to fix the `MPI_BCAST` arguments?

Definition of argument:
https://github.com/impact-lbl/IMPACT-T/blob/cf81f06caa43d1d20ec784bfd9d8427df9b66647/src/DataStruct/Data.f90#L71-L72

My Fortran is very rusty, what goes in here? Does this need to be casted or something?

Some discussion if this is in OpenMPI/MPICH or GCC 10:
https://stackoverflow.com/questions/63892055/fortran-error-type-mismatch-between-two-unrelated-subroutine-calls

**Comment by ChristopherMayes** [comment]
@qianglbl, do you know? I got the suggestion from: https://github.com/PrincetonUniversity/STELLOPT/issues/140

File: README.md (hunk 1 above)
Comment (ax3l) [reply README.md:2]: We just keep PRs concise and fix one thing at a time. That way, they get reviewed faster and one can `git cherry-pick` the changes around.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -13,6 +13,15 @@ project(ImpactT LANGUAGES Fortran)
...
 
+# Allow type mismatch for GCC 10
+# (Adapted from from: https://github.com/Unidata/netcdf-fortran/blob/main/CMakeLists.txt)
+# Check to see if compiler supports
+# -fallow-argument-mismatch-flag introduced in gcc 10.
+include(CheckFortranCompilerFlag)
+check_fortran_compiler_flag("-fallow-argument-mismatch" COMPILER_HAS_ALLOW_ARGUMENT_MISMATCH)
+if(COMPILER_HAS_ALLOW_ARGUMENT_MISMATCH)
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch")
+endif(COMPILER_HAS_ALLOW_ARGUMENT_MISMATCH)
```
Comment (ax3l) [inline src/CMakeLists.txt]: Please move this down into the `if(USE_MPI)` block and use:
```cmake
target_compile_options(${OUTPUT_NAME} PRIVATE -fallow-argument-mismatch)
```
instead of `CMAKE_Fortran_FLAGS`.

Does this also warn with No-MPI builds using our MPI stubs? In that case, add it generally down below the `if(USE_MPI) else` blocks. If not, then only for MPI builds, please (since this is a great check, when its working).

Comment (ChristopherMayes) [reply src/CMakeLists.txt]: done

**Review by ax3l — APPROVED** [review]
Thanks a lot, that looks good to me! :)

File: src/CMakeLists.txt (hunk 3 above)
Comment (ax3l) [reply src/CMakeLists.txt]: Thanks! Does the no-MPI version compile without `-fallow-argument-mismatch` on GCC 10 for you?

Comment (ChristopherMayes) [reply src/CMakeLists.txt]: Yes, works fine, but many warnings: https://github.com/impact-lbl/IMPACT-T/issues/20

**Review by qianglbl — COMMENTED** [review]
Did you test this function?

**Comment by ax3l** [comment]
@ChristopherMayes I think that question is for you :)

@qianglbl generally, we added continuous integration (CI) to IMPACT-T, so at least that one was green.
https://github.com/impact-lbl/IMPACT-T/blob/master/.github/workflows/CI.yml
